pkg/lexer/lexer.go

```diff
@@ -1,5 +1,208 @@
 package lexer
 
+import (
+	"strings"
+	"unicode"
+)
+
 func Run(input string) string {
 	return "Lexing:\n" + input
 }
+
+type Lexer struct {
+	input []rune
+	r     rune // current rune under examination
+	pos   int  // current position in input (points to current rune)
+	line  int
+	col   int
+}
+
+func New(input string) *Lexer {
+	return &Lexer{input: []rune(input), pos: -1, col: -1}
```

Comment:
I think `col` should be `0` - it is reinitialised to zero at the start of every line, and columns should probably count from 1 like lines.